strawberry/object_type.py

```diff
@@ -135,17 +135,19 @@ def _process_type(
     description: Optional[str] = None,
     directives: Optional[Sequence[object]] = (),
     extend: bool = False,
+    extension: Optional[TypeExtension] = None,
     original_type_annotations: Optional[Dict[str, Any]] = None,
 ) -> T:
     name = name or to_camel_case(cls.__name__)
+    extension = extension or TypeExtension()
     original_type_annotations = original_type_annotations or {}
 
     interfaces = _get_interfaces(cls)
-    fields = _get_fields(cls, original_type_annotations)
+    fields = _get_fields(cls, extension, original_type_annotations)
     is_type_of = getattr(cls, "is_type_of", None)
     resolve_type = getattr(cls, "resolve_type", None)
 
-    cls.__strawberry_definition__ = StrawberryObjectDefinition(
+    cls.__strawberry_definition__ = extension.create_object_definition(
```

Comment:
I'm not sure about letting the extension influence the instantiation of the strawberry object definition. This severely limits our ability to add new parameters to the constructor later on without breaking the public API. Addtionally, this implies that only one type extension is supported on a type.

However, having the extension change all of the fields manually after creation could also lead to worse DevX. Open for other opinions on this /cc @strawberry-graphql/core